src/Algebra/Ordered/Structures.agda

```diff
@@ -0,0 +1,205 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Ordered algebraic structures (not packed up with sets, orders,
+-- operations, etc.)
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via
+-- `Algebra.Ordered`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Ordered.Structures where
+
+open import Algebra.Core
+open import Algebra.Definitions
+open import Algebra.Structures
+open import Algebra.Ordered.Definitions using (Monotonic₂)
+open import Data.Product using (proj₁; proj₂)
+open import Level using (_⊔_)
+open import Relation.Binary.Core using (Rel; _⇒_)
+open import Relation.Binary.Definitions using (Transitive)
+open import Relation.Binary.Structures using (IsPreorder; IsPartialOrder)
+
+------------------------------------------------------------------------
+-- Preordered structures
+
+module _ {a ℓ₁ ℓ₂} {A : Set a}
+  (_≈_ : Rel A ℓ₁)  -- The underlying equality.
```

Comment:
Hmm so the equality operator should definitely be made a parameter of the top-level module so you can open it with all the equalities instantiated. I'm trying to figure out whether the order should also be lifted to the top-level as well, but I think it probably shouldn't be as it's more like the algebraic operators whose properties vary between structures.

Reply:
You're right. I just had a look at `Relation.Binary.Structures` and there the equality is a module parameter while the orders are not. Then again, there are some structures there that don't have an order part. Maybe it would make sense to parametrize over everything here. One issue is that different notation are used in `Relation.Binary.Structures` for preorders (`_∼_`)  and partial orders (`_≤_`). This seems a purely cosmetic point, but I tried to be consistent here. If you're fine with just using `_≤_` everywhere, I think it would make sense to make that relation a module parameter too.

Reply:
Yup, go for it! Can't think of any compelling reason not to.

Reply:
Done.